connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -545,6 +545,73 @@ else if (!configState.connectors().contains(connName))
     }
 
     @Override
+    public synchronized void restartConnector(final String connName, final Callback<Void> callback) {
+        addRequest(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                // if config is up to date, we can check if this is a valid connector task. otherwise, we let the
+                // request either be handled locally or forwarded to its target
+                if (isConfigSynced() && !configState.connectors().contains(connName)) {
```

Comment:
Is the behavior when config is not synced what we want? I'm a bit torn on this. On the one hand, it seems like if the config is not synced then we might not be able to properly do anything and might just want to acknowledge that. On the other hand, that seems unfortunate behavior if during that rebalance process APIs just won't work and will fail immediately since they don't think they can reliably forward.

Thoughts?